resources/definitions/anycubic_i3_mega_dual.def.json

```
@@ -0,0 +1,127 @@
+        "jerk_topbottom": { "value": "math.ceil(jerk_print * 5 / 25)" },
+        "jerk_wall": { "value": "math.ceil(jerk_print * 10 / 25)" },
+        "jerk_wall_0": { "value": "math.ceil(jerk_wall * 5 / 10)" },
+        "layer_height_0": { "value": "round(machine_nozzle_size / 1.5, 2)" },
+        "layer_start_x": { "value": "sum(extruderValues('machine_extruder_start_pos_x')) / len(extruderValues('machine_extruder_start_pos_x'))" },
+        "layer_start_y": { "value": "sum(extruderValues('machine_extruder_start_pos_y')) / len(extruderValues('machine_extruder_start_pos_y'))" },
+        "line_width": { "value": "machine_nozzle_size" },
+        "machine_min_cool_heat_time_window": { "value": "15" },
+
+        "prime_tower_enable": { "value": "True" },
+        "raft_airgap": { "value": "0.3" },
+        "raft_base_speed": { "value": "20" },
+        "raft_base_thickness": { "value": "0.12" },
+        "raft_jerk": { "value": "jerk_layer_0" },
+
+        "raft_speed": { "value": "25" },
+        "raft_surface_layers": { "value": "1" },
+        "retraction_amount": { "value": "2" },
+
+        "retraction_hop": { "value": "2" },
+        "retraction_hop_enabled": { "value": "extruders_enabled_count > 1"  },
+        "retraction_hop_only_when_collides": { "value": "True" },
+        "retraction_min_travel": { "value": "2" },
+        "retraction_prime_speed": { "value": "40" },
+
+        "speed_layer_0": { "value": "20" },
+        "speed_prime_tower": { "value": "speed_topbottom" },
+        "speed_print": { "value": "35" },
+        "speed_support": { "value": "speed_wall_0" },
+        "speed_support_interface": { "value": "speed_topbottom" },
+        "speed_topbottom": { "value": "math.ceil(speed_print * 20 / 35)" },
+        "speed_wall": { "value": "math.ceil(speed_print * 30 / 35)" },
+        "speed_wall_0": { "value": "math.ceil(speed_wall * 20 / 30)" },
+        "speed_wall_x": { "value": "speed_wall" },
+        "support_angle": { "value": "55" },
+        "support_pattern": { "value": "'triangles'" },
+        "support_use_towers": { "value": "False" },
+        "support_xy_distance": { "value": "wall_line_width_0 * 2.5" },
+        "support_xy_distance_overhang": { "value": "wall_line_width_0" },
+        "support_z_distance": { "value": "0" },
+
+        "top_bottom_thickness": { "value": "1.2" },
+        "travel_avoid_supports": { "value": "True" },
+        "travel_avoid_distance": { "value": "3 if extruders_enabled_count > 1 else machine_nozzle_tip_outer_diameter / 2 * 1.5" },
+        "wall_0_inset": { "value": "0" },
+        "wall_line_width_x": { "value": "round(line_width * 0.3 / 0.35, 2)" },
+        "wall_thickness": { "value": "line_width * 3" },
+
+        "optimize_wall_printing_order": { "value": "True" },
+        "retraction_combing": { "default_value": "all" },
+        "initial_layer_line_width_factor": { "value": "120" },
+        "zig_zaggify_infill": { "value": "gradual_infill_steps == 0" },
```

```suggestion
        "zig_zaggify_infill": { "value": "gradual_infill_steps == 0" }
```
With the comma this is invalid JSON. Please actually test if the machine loads before contributing it to us.

---

The machine is loaded but without variants. With placing variants,  stops loading and I can not understand where the problem is.

---

Maybe we can help if you show the error message you were getting?